### PR TITLE
feat: Add SliderGraph.ts and sample

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

@GordonSmith this is more of a POC... looking to get your sign-off before spending time polishing the details.

This allows us to treat the Graph as if it was a standard widget from the marshaller's perspective (flat data source)

### CODE REVIEW

**Comment by jbrundage** [comment]
image

**Comment by GordonSmith** [comment]
@jbrundage bump

**Comment by jbrundage** [comment]
Here's a preview. The vertex background color uses a `rainbow` palette and is tied to the filterProperty value:
slidergraph

File: packages/composite/src/SliderGraph.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
+import { HTMLWidget, Palette } from "@hpcc-js/common";
+import { Slider } from "@hpcc-js/form";
+import { Edge, Graph, Vertex } from "@hpcc-js/graph";
+import { Layered } from "@hpcc-js/layout";
+import { select as d3Select } from "d3-selection";
+import { timeFormat as d3TimeFormat } from "d3-time-format";
+
+export class SliderGraph extends HTMLWidget {
```
[the remark this answers is not in the document]
Comment (jbrundage) [reply packages/composite/src/SliderGraph.ts]: Nice, sounds good to me

File: packages/composite/src/SliderGraph.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            });
+        };
+    }
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        d3Select(domNode.parentNode)
+            .style("position", "relative")
+            .style("height", "100%")
+            .style("width", "100%")
+            ;
```
[the remark this answers is not in the document]
Comment (jbrundage) [reply packages/composite/src/SliderGraph.ts]: In that case I will have to do some digging to understand what causes the injected div to require these styles. It's either a bug with extending `HTMLWidget` or I'm just not using it correctly.

File: demos/gallery/samples/widget/SliderGraph.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        [[4,0],{"text":"Sarah Williams","date":"2003-10-19"}],
+        [[9,0],{"text":"Jill Brown","date":"2016-12-26"}],
+        [[4,6,0],{"text":"Bill Brown","date":"2012-11-12"}],
+        [[],{"text":"Jane Williams","date":"2003-11-10"}],
+        [[6,9],{"text":"Chris Davis","date":"2014-12-12"}],
+        [[8],{"text":"Bill Davis","date":"2005-11-21"}],
+        [[],{"text":"Jane Smith","date":"2017-10-22"}],
+        [[7,3,4,9,0,5],{"text":"Jane Johnson","date":"2003-11-20"}],
+        [[2,0,3,1],{"text":"Sarah Jones","date":"2005-10-17"}]
+    ])
```
[the remark this answers is not in the document]
Comment (jbrundage) [reply demos/gallery/samples/widget/SliderGraph.js]: I'll adapt it to use AdjacencyGraph's format

**Comment by GordonSmith** [comment]
@jbrundage bump

**Pull request closed without merging**